patterns/direct_replier.py

```diff
@@ -0,0 +1,122 @@
+## Goal: A simpe request pulblisher to publish a request with reply_to set to any topic.
+# Unlike regular publisher, this will be built on RequestReplyMessagePublisher which exposes the ability to set 'Reply To' field
+import os
+import platform
+import time
+import calendar;
+
+# Import Solace Python  API modules from the solace package
+from solace.messaging.messaging_service import MessagingService, ReconnectionListener, ReconnectionAttemptListener, \
+            ServiceInterruptionListener, RetryStrategy, ServiceEvent
+from solace.messaging.resources.topic import Topic
+from solace.messaging.resources.topic_subscription import TopicSubscription
+from solace.messaging.receiver.request_reply_message_receiver import RequestMessageHandler, InboundMessage, Replier
+
+if platform.uname().system == 'Windows': os.environ["PYTHONUNBUFFERED"] = "1" # Disable stdout buffer 
+
+# Goal is to demonstrate request-reply pattern using Direct messages. A request message carrying {name} is published,
+# and a response with a "Greetings {name}" is expected as response.
+#   direct_requestor: publishes('solace/samples/python/direct/hello/{name}, and waits for response (either synchronously or asynchronously)
+#   direct_replier: subscribes('solace/samples/python/direct/hello/>'), and replies with a greetings message in the body.
+
+TOPIC_PREFIX = "solace/samples/python"
+
+# Handle received messages
+class RequestMessageHandlerImpl(RequestMessageHandler):
+    def on_message(self, request: InboundMessage, replier: Replier):
+        name = request.get_payload_as_string().split("My name is ")[1]
+
+        print(f'============================\n')
+        print(f'<<<<<<<<<<<<<<<<<<<<<<<<<<<<')
+        print(f'Received request (body):' + request.get_payload_as_string())
+        print(f'----------------------------')
+        print(f'Received request:\n{request}')
+
+        message_id = request.get_application_message_id();
+        if replier is not None:
+            outbound_msg = outbound_msg_builder \
+                            .with_application_message_id(f'{message_id}')\
+                            .build(f'Greetings {name}')
+            replier.reply(outbound_msg)
+            print(f'>>>>>>>>>>>>>>>>>>>>>>>>>>>>')
+            print(f'Replied with response (body):\n{outbound_msg.get_payload_as_string()}')
+            print(f'----------------------------')
+            print(f'Replied with response:\n{outbound_msg}')
+        else:
+            print(f'Invalid request, reply_to not set')    
+        print(f'============================')
+
```

Comment:
Made some modifications to the RequestMessageHandlerImpl, what are your thoughts?

```
class RequestMessageHandlerImpl(RequestMessageHandler):
    def __init__(self, message_builder):
        self.message_builder = message_builder

    def on_message(self, request: InboundMessage, replier: Replier):
        request_payload_str = request.get_payload_as_string() if request.get_payload_as_string() != None else request.get_payload_as_bytes()
        if isinstance(request_payload_str, bytearray):
            print(f"Received a message of type: {type(request_payload_str)}. Decoding to string")
            request_payload_str = request_payload_str.decode()

        response_body = f'Greetings {request_payload_str.split("My name is ")[1]}!' if "My name is " in request_payload_str else "Response body"

        print(f'Received request: {request_payload_str}')
        # print(f'Received request message dump: {request}')

        message_id = request.get_application_message_id()
        if replier is not None:
            outbound_msg = self.message_builder \
                            .with_application_message_id(f'{message_id}')\
                            .build(f'{response_body}')
            replier.reply(outbound_msg)
            print(f'>>>>>>>>>>>>>>>>>>>>>>>>>>>>')
            print(f'Replied with response (body): {outbound_msg.get_payload_as_string()}')
            print(f'----------------------------')
            print(f'Response Message dump: {outbound_msg}')
        else:
            print(f'Invalid request, reply_to not set')    
        print(f'============================')
```

Note that this means that the RequestMessageHandlerImpl is called as follows
```
direct_replier.receive_async(RequestMessageHandlerImpl(outbound_msg_builder))
```

Reply:
Incorporated.